src/chesster.js

```diff
@@ -118,6 +118,169 @@ function leagueDMResponse(patterns, responseName) {
 leagueResponse(['captain guidelines'], 'formatCaptainGuidelinesResponse');
 leagueDMResponse(['captains', 'captain list'], 'formatCaptainsResponse');
 
+/* 
+ * strip components
+ * Splits the text. Adds each word to an object using the keys provided. 
+ *
+ * Each value, component, is made to be lower case.
+ *
+ * If the number of components is fewer than the number of words in the text,
+ * the remaining words will be joined as the final argument.
+ */
+function stripComponents(text, componentKeys){
+    var numComponents = componentKeys.length;
+    var toStrip = _.toLower(text);
+
+    var tokens = _.split(toStrip, " ");
+    var args = _.slice(tokens, 0, componentKeys.length-1);
+    var lastArg = _.slice(tokens, componentKeys.length-1).join(" ");
+
+    if( args.length > numComponents ) 
+        throw new Error("Too few arguments to strip");
+
+    var components = {}; //add each argument to the object, using the keys given
+    _.each(args, function(argument, index){
+        components[componentKeys[index]] = tokens[index];
+    });
+    components[componentKeys[numComponents-1]] = lastArg;
+    
+    return components;
+}
+
+/* availability */
+/* [player <player-name> is] {available, unavailable} for round <round-number> in <league> */
+chesster.hears({
+    middleware: [ 
+        slack.withLeague
+    ],
+    patterns: [
+        'available',
+        'unavailable'
+    ],
+    messageTypes: [
+        'direct_message',
+        'direct_mention'
+    ]
+}, 
+function(bot, message){
+    var heltourOptions = message.league.options.heltour;
+    if(!heltourOptions){
+        winston.error("{} league doesn't have heltour options!?".format(message.league.options.name));
+        return;
+    }
+
+    // I hope I am not over-using the ternary operator here. 
+    // I would like to apply stripComponents to the other functions.
+    // but only if we agree this cleans things up.
+    var components = 
+        (! message.text.includes("player")) ? 
+        stripComponents(
+            message.text, 
+            [ 'available', 'for', 'round', 'roundNumber', 'in','league' ]
```

Comment:
This is much better than what we had before, but I think we could go one step further and do something like: 
```
 [ 'available', 'for', 'round', '{roundNumber}', 'in','{league}' ]
```
or
```
 [ 'player', '{playerName}', 'is' 'available', 'for', 'round', '{roundNumber}', 'in','{league}' ]
```

In this case, the { denotes that it's a variable and will take whatever token appears there, while the absence of "{" will indicate that it _must_ be that value and it can throw an error if it's not.

We could take it further and use '|' for two options that can appear at the same spot.

Also, the function could take multiple lists instead of just one and return the first one that it finds that matches, which would remove the need for the ternary operator here. 

Perhaps I'm making the function too complex, but I think that at least the initial idea (denoting which are variable and which are constants) will help